source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,315 @@
+
+       public List<string> Hobbies {get; set;}
+   }
+
+You can also specify a discriminator when registering a class map as follows:
+
+.. code-block:: csharp
+
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.SetDiscriminator("personClass");
+   });
+
+.. tip:: 
+
+   The driver stores your discriminator in the BSON document after the ``_id``
+   field with the name ``_t``.
+
+.. TODO: Link to page on polymorphism/discriminators
+
+Mapping with Constructors
+-------------------------
+
+By default the {+driver-short+} requires a constructor with no arguments in
+order to map classes. However, you can use attributes or expressions in your class
+map registration code to allow the driver to map documents to a constructor with
+arguments.
+
+The following example shows how to customize your constructor using attributes:
+
+.. code-block:: csharp
+   :emphasize-lines: 9
+
+   public class Person 
+   {
+       public string Name { get; set; }
+
+       public int Age { get; set; }
+
+       public List<string> Hobbies {get; set;}
+
+       [BsonConstructor]
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+.. tip:: Multiple BsonConstructors
+
+   If there are more than one constructors with the ``BsonConstructor``
+   attribute, the driver uses the constructor that has the most
+   parameters with matching fields in the document.
+
+The following example shows how map documents to your constructor while
+registering a class map:
+
+.. code-block:: csharp
+   :emphasize-lines: 19
+
+   public class Person 
+   {
+       public string Name { get; set; }
+
+       public int Age { get; set; }
+
+       public List<string> Hobbies {get; set;}
+
+       public Person(string name, string age)
+       {
+           Name = name;
+           Age = age;
+       }
+   }
+
+   BsonClassMap.RegisterClassMap<Person>(classmap =>
+   {
+       classmap.AutoMap();
+       classmap.MapCreator(person => new Person(person.Name, person.Age));
+   });
+
+Customize Class Properties
+--------------------------
+
+You can customize how the driver serializes specific properties in a class by
+using attributes on the property, or by calling methods while registering
+a class map. To learn how to customize properties in your {+language+} class,
+see :ref:`csharp-custom-serialization`.
+
+Dynamically Serialize Properties
+--------------------------------
+
+You can use a method to determine whether or not to serialize a property. For
+the driver to automatically use the method when serializing, the method must be
+named ``ShouldSerialize<property name>``. When the driver sees a method with
+this naming convention, it will use the method to determine whether or not to
+serialize properties of the provided property name.
+
+The following example creates a method that only serializes the ``Age`` property
+if its value is greater than ``0``:
```

Comment:
What happens if the age is <0? Is it not included in the BSON representation at all or set to `null`?

Reply:
It will ignore it if the age is < 0. I added that to the text as well